vmdb/app/controllers/ops_controller/settings/common.rb

```diff
@@ -280,7 +280,13 @@ def settings_update_save
       end
       @edit[:new][:authentication][:ldaphost].reject!(&:blank?) if @edit[:new][:authentication][:ldaphost]
       @changed = (@edit[:new] != @edit[:current].config)
-      @update = MiqServer.find(@sb[:selected_server_id]).get_config("vmdb")
+      server = MiqServer.find(@sb[:selected_server_id])
+      zone = Zone.find_by_name(@edit[:new][:server][:zone])
+      unless zone.nil? || server.zone == zone
+        server.zone = zone
+        server.save
+      end
+      @update = server.get_config("vmdb")
     when "settings_workers"                                   # Workers Settings
```

Comment:
@imtayadeway This looks like the correct change. It seems that it was updating to the new zone in the configuration but never updating the association between the server and the zone objects. I recall that originally the zone was only kept in the configuration and then later on we created a formal association. However, I'm surprised that this bug exists so much time has passed.